src/converters/toZigbee.ts

```diff
@@ -955,44 +955,84 @@ export const light_colortemp_step: Tz.Converter = {
+        const payload: KeyValueAny = {minimum: 0, maximum: 600};
+        let rate: number;
+        let movemode: number;
+
+        // Handle different input formats
+        if (value === "stop") {
+            // String stop command
+            rate = 1;
+            movemode = 0;
+        } else if (utils.isNumber(value)) {
+            // Simple number input
+            const numValue = Number(value);
+            if (numValue === 0) {
+                // Stop command via 0 rate
+                rate = 1;
+                movemode = 0;
             } else {
-                utils.assertNumber(value, key);
-                payload.rate = Math.abs(value);
-                payload.movemode = value > 0 ? 1 : 3;
+                // Normal movement
+                rate = Math.abs(numValue);
+                movemode = numValue > 0 ? 1 : 3;
             }
+        } else if (utils.isObject(value)) {
+            // Object input with rate and optional constraints
+            utils.assertObject(value, key);
 
-            await entity.command("lightingColorCtrl", "moveColorTemp", payload, utils.getOptions(meta.mapped, entity));
+            if (value.rate == null) {
+                throw new Error(`${key}: object must contain 'rate' property`);
+            }
 
-            // We cannot determine the color temperaturefrom the current state so we read it, because
-            // - Color mode could have been switched (x/y or colortemp)
-            if (value === "stop" || value === 0) {
-                const entityToRead = utils.getEntityOrFirstGroupMember(entity);
-                if (entityToRead) {
-                    await utils.sleep(100);
-                    await entityToRead.read("lightingColorCtrl", ["colorTemperature", "colorMode"]);
-                }
+            const rateValue = Number(value.rate);
+            utils.assertNumber(rateValue, `${key}.rate`);
+
+            if (rateValue === 0) {
+                // Stop command via 0 rate
+                rate = 1;
+                movemode = 0;
+            } else {
+                // Normal movement
+                rate = Math.abs(rateValue);
+                movemode = rateValue > 0 ? 1 : 3;
+            }
+
+            // Apply custom constraints if provided
+            if (value.minimum != null) {
+                const minValue = Number(value.minimum);
+                utils.assertNumber(minValue, `${key}.minimum`);
+                payload.minimum = minValue;
+            }
+
+            if (value.maximum != null) {
+                const maxValue = Number(value.maximum);
+                utils.assertNumber(maxValue, `${key}.maximum`);
+                payload.maximum = maxValue;
+            }
+
+            // Validate constraints
+            if (payload.minimum >= payload.maximum) {
+                throw new Error(`${key}: minimum (${payload.minimum}) must be less than maximum (${payload.maximum})`);
             }
         } else {
```

Comment:
I think this branch allowed for a `{"color_temp_move": "up"}`, does that still work?

Reply:
I guess technically this is a breaking change as previously any string that was not one of "stop", "release", "0", "up", "1" would be processed as "down" but now only "down" will be processed, and any other string will raise an error. The previous behavior was inconsistent but I'm happy to re-implement an exact replica of that behavior if desired.

Reply:
That's fine!